BraveShared/Preferences.swift

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+/// An empty protocol simply here to force the developer to use a user defaults encodable value via generic constraint
+public protocol UserDefaultsEncodable {}
+
+/// The applications preferences container
+///
+/// Properties in this object should be of the the type `Option` with the object which is being
+/// stored to automatically interact with `UserDefaults`
+public class Preferences {
+    /// The default `UserDefaults` that all `Option`s will use unless specified
+    public static let defaultContainer = UserDefaults(suiteName: AppInfo.sharedContainerIdentifier)!
+}
+
+/// Defines an object which may watch a set of `Preference.Option`s
+/// - note: @objc was added here due to a Swift compiler bug which doesn't allow a class-bound protocol
+/// to act as `AnyObject` in a `AnyObject` generic constraint (i.e. `WeakList`)
+@objc public protocol PreferencesObserver: class {
+    /// A preference value was changed for some given preference key
+    func preferencesDidChange(for key: String)
+}
+
+extension Preferences {
+    struct DAU {
```

Comment:
`ClientPreferences` is just a wrapper file around `Preferences` which is now defined in `BraveShared`, since it is all just extensions around it, I figured I'd just place in this file for now. If we end up having more than two structs of preferences, we should move it then IMO.